app/controllers/batch_controller.rb

```diff
@@ -32,5 +36,10 @@ def initialize_fields(file)
     file.initialize_fields
   end
 
+  def uploading_on_behalf_of? batch
+    return false if batch.generic_files.count == 0 or batch.generic_files[0].on_behalf_of.blank?
```

Comment:
Doing `batch.generic_files.count` will cause an extra count query to solr.  Instead can we do this?
```ruby
exemplar = batch.generic_files.first
return false if exemplar.nil? || exemplar.on_behalf_of.blank?
current_user.user_key != exemplar.on_behalf_of
```